components/profile.js

```diff
@@ -103,15 +104,283 @@ SteamCommunity.prototype.editProfile = function(settings, callback) {
+												values["rgShowcaseConfig[7][" + n + "][publishedfileid]"] = settings[i][type]["values"][n];
+											}
+										}
+									}
+									break;
+
+								case 'group':
+									values["profile_showcase[9]"] = 9;
+
+									if (settings[i][type].hasOwnProperty("values")) {
+										if (typeof settings[i][type]["values"] === 'object' && settings[i][type]["values"].getSteamID64) {
+											values["rgShowcaseConfig[9][0][accountid]"] = settings[i][type]["values"].getSteamID64();
+										} else {
+											values["rgShowcaseConfig[9][0][accountid]"] = new SteamID(settings[i][type]["values"]).getSteamID64();
+										}
+									}
+									break;
+
+								case 'review':
+									values["profile_showcase[10]"] = 10;
+
+									if (settings[i][type].hasOwnProperty("values")) {
+										values["rgShowcaseConfig[10][0][appid]"] = settings[i][type]["values"];
+									}
+									break;
+
+								case 'workshop':
+									values["profile_showcase[11]"] = 11;
+
+									if (settings[i][type].hasOwnProperty("values")) {
+										values["rgShowcaseConfig[11][0][appid]"] = settings[i][type]["values"]["appid"];
+										values["rgShowcaseConfig[11][0][publishedfileid]"] = settings[i][type]["values"]["publishedfileid"];
+									}
+									break;
+
+								case 'guide':
+									values["profile_showcase[15]"] = 15;
+
+									if (settings[i][type].hasOwnProperty("values")) {
+										values["rgShowcaseConfig[15][0][appid]"] = settings[i][type]["values"]["appid"];
+										values["rgShowcaseConfig[15][0][publishedfileid]"] = settings[i][type]["values"]["publishedfileid"];
+									}
+									break;
+
+								case 'achievements':
+									values["profile_showcase[17]"] = 17;
+
+									if (settings[i][type].hasOwnProperty("values") && settings[i][type]["values"].hasOwnProperty("achievements")) {
+										for (var n = 0; n < 7; n++) {
+											if (settings[i][type]["values"]["achievements"][n] != undefined) {
+												values["rgShowcaseConfig[17][" + n + "][appid]"] = settings[i][type]["values"]["achievements"][n]["appid"];
+												values["rgShowcaseConfig[17][" + n + "][title]"] = settings[i][type]["values"]["achievements"][n]["title"];
+											}
+										}
+									}
+									break;
+
+								case 'games':
+									values["profile_showcase[2]"] = 2;
+
+									if (settings[i][type].hasOwnProperty("values") && settings[i][type]["values"].hasOwnProperty("games")) {
+										showcaseconfig["supplied"] = true;
+										showcaseconfig["numberofrequests"] = 4;
+										showcaseconfig["showcasetype"] = 2;
+										showcaseconfig["itemarray"] = settings[i][type]["values"]["games"];
+									}
+									break;
+
+								case 'ownguides':
+									values["profile_showcase[16]"] = 16;
+
+									if (settings[i][type].hasOwnProperty("values")) {
+										for (var n = 0; n < 4; n++) {
+											if (settings[i][type]["values"][n] != undefined) {
```

Comment:
This is not the proper way to check if something is `undefined` in JavaScript. Otherwise you risk an error. Use:

`if (typeof foo === 'undefined')`

https://stackoverflow.com/a/4726198/5737136

Reply:
The current way wouldn't cause errors as far as I see it and also allows f.e. `null` instead of undefined too.  Even when using `variable !== undefined` the variable will always be undefined or defined, because it is a property of an object. Or can you give a example where this specific code would throw an error?

Reply:
You're right, this shouldn't cause any errors as-is. I think convention is to usually use `typeof`, though, but it's fine as it is now.